File: androidApp/src/main/kotlin/zinqq/app/nav/TabBar.kt
package zinqq.app.nav

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.widthIn
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import zinqq.app.R
import zinqq.app.theme.ZinqqDimens
import zinqq.app.theme.ZinqqTheme

/**
 * The PWA's `TabBar` (U13, KTD-11, R12): a fixed 64dp field-colored bar shown
 * ONLY on Home and Activity — scan icon, WALLET pill, ACTIVITY pill, menu
 * icon (→ Settings). Pills use the uppercase display font; the active pill
 * inverts to `tabActive`/`onTabActive`.
 */
@Composable
fun TabBar(
    current: Route,
    onNavigate: (Route) -> Unit,
) {
    val colors = ZinqqTheme.colors
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .height(ZinqqDimens.TabBarHeight)
            .background(colors.field)
            .padding(horizontal = 8.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.SpaceBetween,
    ) {
        IconButton(
            onClick = { onNavigate(Route.Scan) },
            modifier = Modifier
                .size(ZinqqDimens.MinTouchTarget)
                .clip(CircleShape),
        ) {
            Icon(
                painter = painterResource(R.drawable.ic_scan),
                contentDescription = "Scan QR code",
                tint = colors.onField,
                modifier = Modifier.size(22.dp),
            )
        }

        TabPill(
            label = "Wallet",
            active = current == Route.Home,
            onClick = { onNavigate(Route.Home) },
            modifier = Modifier.weight(1f),
        )

        TabPill(
            label = "Activity",
            active = current == Route.Activity,
            onClick = { onNavigate(Route.Activity) },
            modifier = Modifier.weight(1f),
        )

        IconButton(
            onClick = { onNavigate(Route.Settings) },
            modifier = Modifier
                .size(ZinqqDimens.MinTouchTarget)
                .clip(CircleShape),
        ) {
            Icon(
                painter = painterResource(R.drawable.ic_menu),
                contentDescription = "Settings menu",
                tint = colors.onField,
                modifier = Modifier.size(22.dp),
            )
        }
    }
}

@Composable
private fun TabPill(
    label: String,
    active: Boolean,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    val colors = ZinqqTheme.colors
    Box(
        modifier = modifier
            .widthIn(max = 120.dp)
            .height(ZinqqDimens.MinTouchTarget)
            .clip(CircleShape)
            .background(if (active) colors.tabActive else colors.field)
            .clickable(onClick = onClick),
        contentAlignment = Alignment.Center,
    ) {
        Text(
            text = label.uppercase(),
            color = if (active) colors.onTabActive else colors.onFieldMuted,
            fontFamily = ZinqqTheme.fonts.display,
            fontWeight = FontWeight.Bold,
            fontSize = 14.sp,
            letterSpacing = 1.sp,
        )
    }
}

File: iosApp/iosApp/Navigation/TabBar.swift
import SwiftUI

/// The PWA's `TabBar` (U18, KTD-11, R12): a fixed 64pt field-colored bar
/// shown ONLY on Home and Activity — scan icon, WALLET pill, ACTIVITY pill,
/// menu icon (→ Settings). Pills use the uppercase display font; the active
/// pill inverts to `tabActive`/`onTabActive`. All targets are at least 44pt.
struct TabBar: View {
    let current: Route
    let onNavigate: (Route) -> Void

    @Environment(\.zinqqColors) private var colors

    var body: some View {
        HStack(spacing: 0) {
            iconButton(systemName: "qrcode.viewfinder", label: "Scan QR code") {
                onNavigate(.scan)
            }

            TabPill(
                label: "Wallet",
                active: current == .home,
                onTap: { onNavigate(.home) }
            )
            .frame(maxWidth: .infinity)

            TabPill(
                label: "Activity",
                active: current == .activity,
                onTap: { onNavigate(.activity) }
            )
            .frame(maxWidth: .infinity)

            iconButton(systemName: "line.3.horizontal", label: "Settings menu") {
                onNavigate(.settings)
            }
        }
        .padding(.horizontal, 8)
        .frame(height: ZinqqDimens.tabBarHeight)
        .frame(maxWidth: .infinity)
        .background(colors.field.ignoresSafeArea(edges: .bottom))
    }

    private func iconButton(
        systemName: String,
        label: String,
        action: @escaping () -> Void
    ) -> some View {
        Button(action: action) {
            Image(systemName: systemName)
                .font(.system(size: 22))
                .foregroundColor(colors.onField)
                .frame(width: ZinqqDimens.minTouchTarget, height: ZinqqDimens.minTouchTarget)
        }
        .accessibilityLabel(label)
    }
}

private struct TabPill: View {
    let label: String
    let active: Bool
    let onTap: () -> Void

    @Environment(\.zinqqColors) private var colors

    var body: some View {
        Button(action: onTap) {
            Text(label.uppercased())
                .font(ZinqqFont.display(14, weight: .bold))
                .kerning(1)
                .foregroundColor(active ? colors.onTabActive : colors.onFieldMuted)
                .frame(maxWidth: 120)
                .frame(height: ZinqqDimens.minTouchTarget)
                .background(active ? colors.tabActive : colors.field)
                .clipShape(Capsule())
        }
        .accessibilityLabel(label)
        .accessibilityAddTraits(active ? .isSelected : [])
    }
}
